datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -754,24 +775,37 @@ mod test {
         \n    TableScan: test";
 
         assert_optimized_plan_eq(expected, &plan);
-
         Ok(())
     }
 
     #[test]
     fn redundant_project_fields() {
         let table_scan = test_table_scan().unwrap();
         let affected_id: BTreeSet<Identifier> =
-            ["c+a".to_string(), "d+a".to_string()].into_iter().collect();
-        let expr_set = [
+            ["c+a".to_string(), "b+a".to_string()].into_iter().collect();
```

Comment:
Some notes for this and the next test case's changes:

The mock table scan doesn't have a column named "`d`", so I changed it to "`b`". And "a", "b" are ambiguous when join, thus I add qualifiers to them.